packages/headless/src/features/facets/facet-set/facet-set-slice.ts

```diff
@@ -86,6 +91,29 @@ export const facetSetReducer = createReducer(
         facetRequest.freezeCurrentValues = true;
         facetRequest.preventAutoSelect = true;
       })
+      .addCase(toggleSingleSelectFacetValue, (state, action) => {
+        const {facetId, selection} = action.payload;
```

Comment:
Here also, there's a lot of copy paste from the `toggleSelectFacetValue` case.

We could also extract a common function for both of them directly in this module.

Just like you did for `handleSelectSearchResult`